File: 프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict

def solution(clothes):
    answer = 1
    clothes_dict = defaultdict(set)
    
    for name, typ in clothes:
        clothes_dict[typ].add(name)
    
    for typ in clothes_dict:
        answer *= len(clothes_dict[typ])+1
    
    return answer-1

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    
    unordered_map<string, unordered_set<string>> clothes_map;
    for (auto& p : clothes){
        clothes_map[p[1]].insert(p[0]);
    }
    
    for (auto& [type, items] : clothes_map){
        answer *= (items.size()+1);
    }
    answer--;
    
    
    /*
    for (auto& c : clothes_map){
        cout << c.second.size() << "\n";
    }
    */
    
    return answer;
}